lesson01/assignment/scripts.js

```diff
@@ -1,26 +1,134 @@
-        input = $('.input'),
-        submit = $('.submit'),
-        resultSum = $('.sum');
-    // Click event on the submit input
-    submit.on('click', function () {
-        // Grab the input value and parse it into a number
-        const val = input.val();
-        const num = parseInt(val);
-        // Verify the user gave us a real number
-        if (!isNaN(num)) {
-            // Add our new number to the data array
-            data.push(num);
-            // Use reduce to convert our array of numbers into a single value - the sum of all of the numbers
-            const sum = data.reduce((acc, curr) => acc + curr);
-            // Add our new total to the page
-            resultSum.text(sum);
-            // Reset the input value to an empty string
-            input.val('');
-        }
-    });
+//     const data = [1, 2, 3, 4],
+//         input = $('.input'),
+//         submit = $('.submit'),
+//         resultSum = $('.sum');
+//     // Click event on the submit input
+//     submit.on('click', function () {
+//         // Grab the input value and parse it into a number
+//         const val = input.val();
+//         const num = parseInt(val);
+//         // Verify the user gave us a real number
+//         if (!isNaN(num)) {
+//             // Add our new number to the data array
+//             data.push(num);
+//             // Use reduce to convert our array of numbers into a single value - the sum of all of the numbers
+//             const sum = data.reduce((acc, curr) => acc + curr);
+//             // Add our new total to the page
+//             resultSum.text(sum);
+//             // Reset the input value to an empty string
+//             input.val('');
+//         }
+//     });
+// });
+//map() method
+
+$(function() {
+const data = [5, 4, 8, 9, 3],
+      input = $('.input'),
+      submit = $('.submit'),
+      result = $('.showOutput');
+      // click is an event here
+ submit.on('click', function() {
+ //input value parsed into number
+     const number = input.val();
+     let num = parseInt(number);
+    // define a variable which holds the functionality of map() method
+
+     const som = data.map(function(n) {
+     //here num is the user input number
+     // n is each element of the array
+     //here the map method takes the input value, multiplies it to each elements of an array and returns a new array
+                  return num * n;
+
+
+       });
+        //console.log(num);
+      result.text(som);
+      input.val('');
+
+});
+});
+
+
+//filter method()
+$(function () {
+const array = [4, 8, 6, 7, 3, 11, 16];
+
+$(".submit1").on("click", function() {
+  //get user input value
+  const inputVal = $(".input1").val();
+  //validate inputvalue to numbers using parseInt
+  const inputVal1 = parseInt(inputVal);
+// use filter function to differentiate numbers which has zero remainder aftervdevided by 2(even numbers in array)
+  let arr2 = array.filter(function(num) {
+  return num % 2 == 0;
+  })
```

Comment:
This is a good use of the 'filter' method, but it doesn't do anything with the user's input. Maybe instead of building an array of the numbers that are divisible by 2, it could build an array of the numbers that are divisible by the user's input number?

Reply:
Much better!